From ea96725b5156135d5875415f75d2188f6f56622a Mon Sep 17 00:00:00 2001
From: Kevin Scholz <k-scholz@ti.com>
Date: Thu, 9 Feb 2023 16:00:53 -0600
Subject: [PATCH 15/15] arm: dts: k3-j721e-ddr-beagleboneai64-lp4-3200: Update
 ODT/drive strengths

Use the latest tool version to 10.1 and the following On Die
Termination(ODT)/drive strength updates:
 - SOC DQ Drive Strength (write): 40 Ohm to 48 Ohm
 - SOC DQ ODT (read): 40 Ohm to 48 Ohm
 - DRAM DQ Drive Strength (read): 40 Ohm to 48 Ohm
 - DRAM DQ ODT (write): 40 Ohm to 48 Ohm (+ enable ODT during boot)
 - DRAM CA ODT: 80 Ohm to 60 Ohm (+ enable ODT during boot)
 - DRAM DQ VREF F0: Match F1 / F2
 - Enable ODT CK Override
 - SOC CA Drive Strength: 40 Ohm to 34.3 Ohm
 - SOC CS Drive Strength: 80 Ohm to 34.3 Ohm

Signed-off-by: Kevin Scholz <k-scholz@ti.com>
Signed-off-by: Nishanth Menon <nm@ti.com>
---
 .../k3-j721e-ddr-beagleboneai64-lp4-3200.dtsi | 106 +++++++++---------
 1 file changed, 53 insertions(+), 53 deletions(-)

diff --git a/arch/arm/dts/k3-j721e-ddr-beagleboneai64-lp4-3200.dtsi b/arch/arm/dts/k3-j721e-ddr-beagleboneai64-lp4-3200.dtsi
index 09445f92c0..4167ac21a8 100644
--- a/arch/arm/dts/k3-j721e-ddr-beagleboneai64-lp4-3200.dtsi
+++ b/arch/arm/dts/k3-j721e-ddr-beagleboneai64-lp4-3200.dtsi
@@ -1,8 +1,8 @@
 // SPDX-License-Identifier: GPL-2.0+
 /*
- * Copyright (C) 2019-2022 Texas Instruments Incorporated - http://www.ti.com/
- * This file was generated by the Jacinto7_DDRSS_RegConfigTool, Revision: 0.9.1
- * This file was generated on 10/21/2022
+ * Copyright (C) 2019 Texas Instruments Incorporated - http://www.ti.com/
+ * This file was generated by the Jacinto7_DDRSS_RegConfigTool, Revision: 0.10.1
+ * This file was generated on 02/08/2023
  * Part Number: Kingston Q3222PM1WDGTK-U
  * Configuration: LPDDR4-3200, wrDBI enabled, j721e-SK latencies
  * Also common for:
@@ -190,20 +190,20 @@
 #define DDRSS_CTL_173_DATA 0x00000000
 #define DDRSS_CTL_174_DATA 0x00000000
 #define DDRSS_CTL_175_DATA 0x2DD40084
-#define DDRSS_CTL_176_DATA 0xB3002DD4
-#define DDRSS_CTL_177_DATA 0x0000B3B3
-#define DDRSS_CTL_178_DATA 0x36000000
-#define DDRSS_CTL_179_DATA 0x27270036
-#define DDRSS_CTL_180_DATA 0x0F0F0000
-#define DDRSS_CTL_181_DATA 0x16000000
-#define DDRSS_CTL_182_DATA 0x00841616
+#define DDRSS_CTL_176_DATA 0xAB002DD4
+#define DDRSS_CTL_177_DATA 0x0000ABAB
+#define DDRSS_CTL_178_DATA 0x45450000
+#define DDRSS_CTL_179_DATA 0x27272745
+#define DDRSS_CTL_180_DATA 0x0F0F0F00
+#define DDRSS_CTL_181_DATA 0x1D000000
+#define DDRSS_CTL_182_DATA 0x00841D1D
 #define DDRSS_CTL_183_DATA 0x2DD42DD4
-#define DDRSS_CTL_184_DATA 0xB3B3B300
+#define DDRSS_CTL_184_DATA 0xABABAB00
 #define DDRSS_CTL_185_DATA 0x00000000
-#define DDRSS_CTL_186_DATA 0x00363600
-#define DDRSS_CTL_187_DATA 0x00002727
+#define DDRSS_CTL_186_DATA 0x27454545
+#define DDRSS_CTL_187_DATA 0x0F002727
 #define DDRSS_CTL_188_DATA 0x00000F0F
-#define DDRSS_CTL_189_DATA 0x16161600
+#define DDRSS_CTL_189_DATA 0x1D1D1D00
 #define DDRSS_CTL_190_DATA 0x00000020
 #define DDRSS_CTL_191_DATA 0x00000000
 #define DDRSS_CTL_192_DATA 0x00000001
@@ -749,30 +749,30 @@
 #define DDRSS_PI_272_DATA 0x00080804
 #define DDRSS_PI_273_DATA 0x00000000
 #define DDRSS_PI_274_DATA 0x00000000
-#define DDRSS_PI_275_DATA 0x00B30084
-#define DDRSS_PI_276_DATA 0x00160000
-#define DDRSS_PI_277_DATA 0x36B32DD4
-#define DDRSS_PI_278_DATA 0x00160F27
-#define DDRSS_PI_279_DATA 0x36B32DD4
-#define DDRSS_PI_280_DATA 0x00160F27
-#define DDRSS_PI_281_DATA 0x00B30084
-#define DDRSS_PI_282_DATA 0x00160000
-#define DDRSS_PI_283_DATA 0x36B32DD4
-#define DDRSS_PI_284_DATA 0x00160F27
-#define DDRSS_PI_285_DATA 0x36B32DD4
-#define DDRSS_PI_286_DATA 0x00160F27
-#define DDRSS_PI_287_DATA 0x00B30084
-#define DDRSS_PI_288_DATA 0x00160000
-#define DDRSS_PI_289_DATA 0x36B32DD4
-#define DDRSS_PI_290_DATA 0x00160F27
-#define DDRSS_PI_291_DATA 0x36B32DD4
-#define DDRSS_PI_292_DATA 0x00160F27
-#define DDRSS_PI_293_DATA 0x00B30084
-#define DDRSS_PI_294_DATA 0x00160000
-#define DDRSS_PI_295_DATA 0x36B32DD4
-#define DDRSS_PI_296_DATA 0x00160F27
-#define DDRSS_PI_297_DATA 0x36B32DD4
-#define DDRSS_PI_298_DATA 0x00160F27
+#define DDRSS_PI_275_DATA 0x45AB0084
+#define DDRSS_PI_276_DATA 0x001D0F27
+#define DDRSS_PI_277_DATA 0x45AB2DD4
+#define DDRSS_PI_278_DATA 0x001D0F27
+#define DDRSS_PI_279_DATA 0x45AB2DD4
+#define DDRSS_PI_280_DATA 0x001D0F27
+#define DDRSS_PI_281_DATA 0x45AB0084
+#define DDRSS_PI_282_DATA 0x001D0F27
+#define DDRSS_PI_283_DATA 0x45AB2DD4
+#define DDRSS_PI_284_DATA 0x001D0F27
+#define DDRSS_PI_285_DATA 0x45AB2DD4
+#define DDRSS_PI_286_DATA 0x001D0F27
+#define DDRSS_PI_287_DATA 0x45AB0084
+#define DDRSS_PI_288_DATA 0x001D0F27
+#define DDRSS_PI_289_DATA 0x45AB2DD4
+#define DDRSS_PI_290_DATA 0x001D0F27
+#define DDRSS_PI_291_DATA 0x45AB2DD4
+#define DDRSS_PI_292_DATA 0x001D0F27
+#define DDRSS_PI_293_DATA 0x45AB0084
+#define DDRSS_PI_294_DATA 0x001D0F27
+#define DDRSS_PI_295_DATA 0x45AB2DD4
+#define DDRSS_PI_296_DATA 0x001D0F27
+#define DDRSS_PI_297_DATA 0x45AB2DD4
+#define DDRSS_PI_298_DATA 0x001D0F27
 #define DDRSS_PI_299_DATA 0x00000000
 
 #define DDRSS_PHY_00_DATA 0x000004F0
@@ -859,8 +859,8 @@
 #define DDRSS_PHY_81_DATA 0x00000000
 #define DDRSS_PHY_82_DATA 0x00000000
 #define DDRSS_PHY_83_DATA 0x00000100
-#define DDRSS_PHY_84_DATA 0x01CC0C01
-#define DDRSS_PHY_85_DATA 0x1003CC0C
+#define DDRSS_PHY_84_DATA 0x01BB0B01
+#define DDRSS_PHY_85_DATA 0x1003BB0B
 #define DDRSS_PHY_86_DATA 0x20000140
 #define DDRSS_PHY_87_DATA 0x07FF0200
 #define DDRSS_PHY_88_DATA 0x0000DD01
@@ -1115,8 +1115,8 @@
 #define DDRSS_PHY_337_DATA 0x00000000
 #define DDRSS_PHY_338_DATA 0x00000000
 #define DDRSS_PHY_339_DATA 0x00000100
-#define DDRSS_PHY_340_DATA 0x01CC0C01
-#define DDRSS_PHY_341_DATA 0x1003CC0C
+#define DDRSS_PHY_340_DATA 0x01BB0B01
+#define DDRSS_PHY_341_DATA 0x1003BB0B
 #define DDRSS_PHY_342_DATA 0x20000140
 #define DDRSS_PHY_343_DATA 0x07FF0200
 #define DDRSS_PHY_344_DATA 0x0000DD01
@@ -1371,8 +1371,8 @@
 #define DDRSS_PHY_593_DATA 0x00000000
 #define DDRSS_PHY_594_DATA 0x00000000
 #define DDRSS_PHY_595_DATA 0x00000100
-#define DDRSS_PHY_596_DATA 0x01CC0C01
-#define DDRSS_PHY_597_DATA 0x1003CC0C
+#define DDRSS_PHY_596_DATA 0x01BB0B01
+#define DDRSS_PHY_597_DATA 0x1003BB0B
 #define DDRSS_PHY_598_DATA 0x20000140
 #define DDRSS_PHY_599_DATA 0x07FF0200
 #define DDRSS_PHY_600_DATA 0x0000DD01
@@ -1627,8 +1627,8 @@
 #define DDRSS_PHY_849_DATA 0x00000000
 #define DDRSS_PHY_850_DATA 0x00000000
 #define DDRSS_PHY_851_DATA 0x00000100
-#define DDRSS_PHY_852_DATA 0x01CC0C01
-#define DDRSS_PHY_853_DATA 0x1003CC0C
+#define DDRSS_PHY_852_DATA 0x01BB0B01
+#define DDRSS_PHY_853_DATA 0x1003BB0B
 #define DDRSS_PHY_854_DATA 0x20000140
 #define DDRSS_PHY_855_DATA 0x07FF0200
 #define DDRSS_PHY_856_DATA 0x0000DD01
@@ -1839,7 +1839,7 @@
 #define DDRSS_PHY_1061_DATA 0x00000000
 #define DDRSS_PHY_1062_DATA 0x00000000
 #define DDRSS_PHY_1063_DATA 0x00000000
-#define DDRSS_PHY_1064_DATA 0x000305CC
+#define DDRSS_PHY_1064_DATA 0x000305FF
 #define DDRSS_PHY_1065_DATA 0x00030000
 #define DDRSS_PHY_1066_DATA 0x00000300
 #define DDRSS_PHY_1067_DATA 0x00000300
@@ -2181,20 +2181,20 @@
 #define DDRSS_PHY_1403_DATA 0x00000390
 #define DDRSS_PHY_1404_DATA 0x00000390
 #define DDRSS_PHY_1405_DATA 0x00000005
-#define DDRSS_PHY_1406_DATA 0x01813FCC
-#define DDRSS_PHY_1407_DATA 0x000000CC
+#define DDRSS_PHY_1406_DATA 0x01813FFF
+#define DDRSS_PHY_1407_DATA 0x000000FF
 #define DDRSS_PHY_1408_DATA 0x0C000DFF
 #define DDRSS_PHY_1409_DATA 0x30000DFF
 #define DDRSS_PHY_1410_DATA 0x3F0DFF11
 #define DDRSS_PHY_1411_DATA 0x000100F0
-#define DDRSS_PHY_1412_DATA 0x780DFFCC
+#define DDRSS_PHY_1412_DATA 0x780DFFFF
 #define DDRSS_PHY_1413_DATA 0x00007E31
 #define DDRSS_PHY_1414_DATA 0x000CBF11
-#define DDRSS_PHY_1415_DATA 0x01990010
+#define DDRSS_PHY_1415_DATA 0x01FF0010
 #define DDRSS_PHY_1416_DATA 0x000CBF11
-#define DDRSS_PHY_1417_DATA 0x01990010
+#define DDRSS_PHY_1417_DATA 0x01FF0010
 #define DDRSS_PHY_1418_DATA 0x3F0DFF11
-#define DDRSS_PHY_1419_DATA 0x00EF00F0
+#define DDRSS_PHY_1419_DATA 0x01FF00F0
 #define DDRSS_PHY_1420_DATA 0x3F0DFF11
 #define DDRSS_PHY_1421_DATA 0x01FF00F0
 #define DDRSS_PHY_1422_DATA 0x20040006
-- 
2.43.0

